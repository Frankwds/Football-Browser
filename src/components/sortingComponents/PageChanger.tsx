import { Box, Center, Select } from "@chakra-ui/react";
import { IconButton } from "@chakra-ui/react";
import "./../style.css";
import {
  AiOutlineArrowLeft,
  AiOutlineArrowRight,
  AiOutlineArrowDown,
  AiOutlineArrowUp,
} from "react-icons/ai";

import { StateType } from "../../redux/StateType";
import { useState } from "react";
import { useDispatch, useSelector } from "react-redux";

type Props = {
  func: any;
  page: number;
  lastpage: number;
  color?: string;
  accent?: string;
  pageSize: number;
};

/**
 * Page changer - Handler for changing which matches are displayed
 *
 * @param func
 * @param color
 * @param accent
 * @param page
 *
 *
 *
 * @returns {React.FC}
 *
 * The func parameter is the callback-function passed on from @link {../containerComponents/Content.tsx}
 */

const PageChanger: React.FC<Props> = ({
  func,
  color,
  accent,
  page,
  lastpage,
  pageSize,
}) => {
  const updateSearchQueries = useDispatch();

  const acc = useSelector(
    (state: StateType) => state.searchQueries.searchQueries.acc
  );
  const league = useSelector(
    (state: StateType) => state.searchQueries.searchQueries.league
  );
  const country = useSelector(
    (state: StateType) => state.searchQueries.searchQueries.country
  );
  const season = useSelector(
    (state: StateType) => state.searchQueries.searchQueries.season
  );
  const ht = useSelector(
    (state: StateType) => state.searchQueries.searchQueries.ht
  );
  const at = useSelector(
    (state: StateType) => state.searchQueries.searchQueries.at
  );

  const btIcon = () => {
    if (acc) {
      return <AiOutlineArrowUp />;
    } else {
      return <AiOutlineArrowDown />;
    }
  };
  const handleApplyFilter = (s: string) => {
    // Provide given search query object
    let a = false;
    if (acc) {
      a = false;
    } else {
      a = true;
    }

    updateSearchQueries({
      type: "UPDATE_SEARCH_DATA",
      payload: {
        league: league,
        country: country,
        season: season,
        ht: ht,
        at: at,
        sort: s,
        acc: a,
      },
    });
  };

  const [value, setValue] = useState("date");

  const handleChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
    let val = event.target.value;
    if (event.target.value == "") {
      val = "date";
    }
    setValue(val);
  };

  return (
    <Box
      fontSize="xl"
      color={accent}
      textAlign={"center"}
      bg={color}
      borderRadius="lg"
      h={"40px"}
    >
      {" "}
      <Center>
        <IconButton
          bg={color}
          color={accent}
          aria-label="Go page back"
          float={"left"}
          icon={<AiOutlineArrowLeft />}
          onClick={() => func(page - 1)}
        />
        <Box w="100%">
          Page: {page + 1} of {Math.ceil(lastpage / pageSize)}
        </Box>
        <Center w="100%">
          <Select value={value} onChange={handleChange} placeholder="Sort by:">
            <option value="date">Date</option>
            <option value="country">Country</option>
            <option value="league">League</option>
          </Select>
          <IconButton
            bg={color}
            color={accent}
            aria-label="Accending"
            icon={btIcon()}
            onClick={() => handleApplyFilter(value)}
          />
        </Center>
        <IconButton
          bg={color}
          color={accent}
          aria-label="Go page front"
          float={"right"}
          icon={<AiOutlineArrowRight />}
          onClick={() => func(page + 1)}
        />
      </Center>
    </Box>
  );
};

export default PageChanger;
