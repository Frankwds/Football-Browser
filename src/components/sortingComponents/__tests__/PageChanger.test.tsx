import ReactDOM from "react-dom";
import { act } from "react-dom/test-utils";
import renderer from "react-test-renderer";
import PageChanger from "../PageChanger";

const col = "rgb(192,192,192)";
const acc = "rgb(0,128,128)";
let container: any;

beforeEach(() => {
  container = document.createElement("div");
  document.body.appendChild(container);
});

afterEach(() => {
  document.body.removeChild(container);
  container = null;
});

describe("Testing PageChanger", () => {
  it("should render without crashing", () => {
    act(() => {
      const div = document.createElement("div");
      ReactDOM.render(
        <PageChanger
          page={0}
          lastpage={10}
          pageSize={5}
          func={() => {}}
          color={col}
          accent={acc}
        />,
        div
      );
    });
  });

  it("snapshot should be same as previous", () => {
    const tree = renderer
      .create(
        <PageChanger
          page={0}
          lastpage={10}
          pageSize={5}
          func={() => {}}
          color={col}
          accent={acc}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
