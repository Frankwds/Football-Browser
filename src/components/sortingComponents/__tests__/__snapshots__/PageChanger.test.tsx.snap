// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing PageChanger snapshot should be same as previous 1`] = `
Array [
  <div
    className="css-cl38x"
  >
     
    <div
      className="css-gmuwbf"
    >
      <button
        aria-label="Go page back"
        className="chakra-button css-1htp875"
        onClick={[Function]}
        type="button"
      >
        <svg
          aria-hidden={true}
          fill="currentColor"
          focusable={false}
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 1024 1024"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M872 474H286.9l350.2-304c5.6-4.9 2.2-14-5.2-14h-88.5c-3.9 0-7.6 1.4-10.5 3.9L155 487.8a31.96 31.96 0 0 0 0 48.3L535.1 866c1.5 1.3 3.3 2 5.2 2h91.5c7.4 0 10.8-9.2 5.2-14L286.9 550H872c4.4 0 8-3.6 8-8v-60c0-4.4-3.6-8-8-8z"
          />
        </svg>
      </button>
      <div
        className="css-8atqhb"
      >
        Page: 
        1
         of 
        2
      </div>
      <div
        className="css-1lads1q"
      >
        <div
          className="chakra-select__wrapper css-42b2qy"
        >
          <select
            className="chakra-select css-1tc12v8"
            id="selectSort"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            value="date"
          >
            <option
              value=""
            >
              Sort by:
            </option>
            <option
              value="date"
            >
              Date
            </option>
            <option
              value="country"
            >
              Country
            </option>
            <option
              value="league"
            >
              League
            </option>
          </select>
          <div
            className="chakra-select__icon-wrapper css-zj2dqk"
          >
            <svg
              aria-hidden={true}
              className="chakra-select__icon"
              focusable={false}
              role="presentation"
              style={
                Object {
                  "color": "currentColor",
                  "height": "1em",
                  "width": "1em",
                }
              }
              viewBox="0 0 24 24"
            >
              <path
                d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                fill="currentColor"
              />
            </svg>
          </div>
        </div>
        <button
          aria-label="Accending"
          className="chakra-button css-1xmih8r"
          id="sortButton"
          onClick={[Function]}
          type="button"
        >
          <svg
            aria-hidden={true}
            fill="currentColor"
            focusable={false}
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 1024 1024"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M862 465.3h-81c-4.6 0-9 2-12.1 5.5L550 723.1V160c0-4.4-3.6-8-8-8h-60c-4.4 0-8 3.6-8 8v563.1L255.1 470.8c-3-3.5-7.4-5.5-12.1-5.5h-81c-6.8 0-10.5 8.1-6 13.2L487.9 861a31.96 31.96 0 0 0 48.3 0L868 478.5c4.5-5.2.8-13.2-6-13.2z"
            />
          </svg>
        </button>
      </div>
      <button
        aria-label="Go page front"
        className="chakra-button css-13f712g"
        onClick={[Function]}
        type="button"
      >
        <svg
          aria-hidden={true}
          fill="currentColor"
          focusable={false}
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 1024 1024"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M869 487.8L491.2 159.9c-2.9-2.5-6.6-3.9-10.5-3.9h-88.5c-7.4 0-10.8 9.2-5.2 14l350.2 304H152c-4.4 0-8 3.6-8 8v60c0 4.4 3.6 8 8 8h585.1L386.9 854c-5.6 4.9-2.2 14 5.2 14h91.5c1.9 0 3.8-.7 5.2-2L869 536.2a32.07 32.07 0 0 0 0-48.4z"
          />
        </svg>
      </button>
    </div>
  </div>,
  <span />,
]
`;
